### PR TITLE
chore(components/progress-indicator): progress indicator strict mode support

### PR DESCRIPTION
**Pull request by Blackbaud-PaulCrowder**

### CODE REVIEW

File: libs/components/progress-indicator/src/index.ts [hunk 1]
```diff
@@ -3,7 +3,6 @@ export * from './lib/modules/progress-indicator/types/progress-indicator-action-
 export * from './lib/modules/progress-indicator/types/progress-indicator-action-click-progress-handler';
 export * from './lib/modules/progress-indicator/types/progress-indicator-change';
 export * from './lib/modules/progress-indicator/types/progress-indicator-display-mode-type';
-export * from './lib/modules/progress-indicator/types/progress-indicator-item-status';
```
Comment (Blackbaud-PaulCrowder) [inline libs/components/progress-indicator/src/index.ts]: I verified this type is never exposed by our public API and is not referenced by any internal applications.

File: libs/components/progress-indicator/src/lib/modules/progress-indicator/types/progress-indicator-item-status-type.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export type SkyProgressIndicatorItemStatusType =
```
Comment (Blackbaud-PaulCrowder) [inline libs/components/progress-indicator/src/lib/modules/progress-indicator/types/progress-indicator-item-status-type.ts]: The code is much cleaner after changing this enum to a string union.

**Comment by Blackbaud-PaulCrowder** [comment]
[AB#2195601](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2195601)
[AB#2195513](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2195513)

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/762/storybook/)

Component Storybooks:

- _(no component storybooks affected in this pr)_ 

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/762/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/762/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/762/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/762/playground/)

File: libs/components/progress-indicator/src/lib/modules/progress-indicator/types/progress-indicator-item-status.ts [hunk 3]
```diff
@@ -1,21 +0,0 @@
-export enum SkyProgressIndicatorItemStatus {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/progress-indicator/src/lib/modules/progress-indicator/types/progress-indicator-item-status.ts]: This file is exported from the public exports API and documented here: https://developer.blackbaud.com/skyux/components/progress-indicator-waterfall?docs-active-tab=design. Should we remove this file in a separate "breaking change" pull request?

File: libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-nav-button/progress-indicator-nav-button.component.ts [hunk 4]
```diff
@@ -141,94 +128,68 @@ export class SkyProgressIndicatorNavButtonComponent
...
-  private _progressIndicator: SkyProgressIndicatorComponent;
+  #changeDetector: ChangeDetectorRef;
+  #parentComponent: SkyProgressIndicatorComponent;
 
   constructor(
-    private changeDetector: ChangeDetectorRef,
-    @Optional() private parentComponent: SkyProgressIndicatorComponent
-  ) {}
+    changeDetector: ChangeDetectorRef,
+    @Optional() parentComponent: SkyProgressIndicatorComponent
```
Comment (Blackbaud-SteveBrush) [inline libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-nav-button/progress-indicator-nav-button.component.ts]: Can we add the `?` to the parameter definition?
```
@Optional() parentComponent?: SkyProgressIndicatorComponent
```

File: libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-nav-button/progress-indicator-nav-button.component.ts [hunk 5]
```diff
@@ -141,94 +128,68 @@ export class SkyProgressIndicatorNavButtonComponent
...
+  #_disabled: boolean | undefined;
+  #_isVisible: boolean | undefined;
+  #_progressIndicator: SkyProgressIndicatorComponent | undefined;
 
-  private _buttonType: SkyProgressIndicatorNavButtonType;
-  private _disabled: boolean;
-  private _isVisible: boolean;
-  private _progressIndicator: SkyProgressIndicatorComponent;
+  #changeDetector: ChangeDetectorRef;
+  #parentComponent: SkyProgressIndicatorComponent;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-nav-button/progress-indicator-nav-button.component.ts]: See comment below, `SkyProgressIndicatorComponent | undefined`.

File: libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-status-marker/progress-indicator-status-marker.component.ts [hunk 6]
```diff
@@ -26,75 +26,44 @@ import { SkyProgressIndicatorItemStatus } from '../types/progress-indicator-item
 export class SkyProgressIndicatorStatusMarkerComponent implements OnDestroy {
   @Input()
   public set displayMode(value: 'vertical' | 'horizontal') {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-status-marker/progress-indicator-status-marker.component.ts:28]: Should be `| undefined`?

File: libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-status-marker/progress-indicator-status-marker.component.ts [hunk 7]
```diff
@@ -26,75 +26,44 @@ import { SkyProgressIndicatorItemStatus } from '../types/progress-indicator-item
...
-
-    return this._displayMode;
+    return this.#_displayMode;
   }
 
   @Input()
-  public set status(value: SkyProgressIndicatorItemStatus) {
-    this._status = value;
-    this.changeDetector.markForCheck();
+  public set status(value: SkyProgressIndicatorItemStatusType) {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-status-marker/progress-indicator-status-marker.component.ts]: Should be `| undefined`?

File: libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-status-marker/progress-indicator-status-marker.component.ts (hunk 6 above)
Comment (Blackbaud-PaulCrowder) [reply libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-status-marker/progress-indicator-status-marker.component.ts:28]: I left off `| undefined` on internal component inputs that are always set to a value.

File: libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-status-marker/progress-indicator-status-marker.component.ts (hunk 7 above)
Comment (Blackbaud-PaulCrowder) [reply libs/components/progress-indicator/src/lib/modules/progress-indicator/progress-indicator-status-marker/progress-indicator-status-marker.component.ts]: Same as above.

File: libs/components/progress-indicator/src/lib/modules/progress-indicator/types/progress-indicator-item-status.ts (hunk 3 above)
Comment (Blackbaud-PaulCrowder) [reply libs/components/progress-indicator/src/lib/modules/progress-indicator/types/progress-indicator-item-status.ts]: No idea what I was thinking. I put the enum back.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**